pytorch_lightning/trainer/trainer.py

```diff
@@ -1404,10 +1411,23 @@ def _call_setup_hook(self) -> None:
 
     def _call_configure_sharded_model(self) -> None:
         with self.accelerator.model_sharded_context():
-            materialize_module(self.lightning_module)
+            self._handle_meta_model()
             self.call_hook("configure_sharded_model")
             self.call_hook("on_configure_sharded_model")
 
+    def _handle_meta_model(self) -> None:
+        param = next(self.lightning_module.parameters())
+        if param.device.type != "meta":
+            return
+
+        if isinstance(self.training_type_plugin, (DDPSpawnPlugin, TPUSpawnPlugin)):
```

Comment:
```suggestion
        if isinstance(self.training_type_plugin, DDPSpawnPlugin):
```
Not required, as TPUSpawn plugin is a subclass of it